cmd/deck/static/configured-jobs/configured-jobs.ts

```diff
@@ -0,0 +1,166 @@
+        dialog.close();
+    });
+}
+
+interface Repo {
+    org: Org;
+    safeName: string;
+    name: string;
+    jobs: Job[];
+}
+
+interface Org {
+    name: string
+}
+
+interface Job {
+    name: string;
+    type: string;
+    yamlDefinition: string;
+    jobHistoryLink: string;
+}
+
+/**
+ * Redraws the content of the page.
+ */
+function redraw(): void {
+    redrawOptions();
+    includedRepos.forEach((repo) => {
+        redrawRepo(repo);
+    })
+}
+
+/**
+ * Redraws the repo selection box
+ */
+function redrawOptions(): void {
+    const repos = allRepos.sort();
+    const sel = document.getElementById("repo") as HTMLSelectElement;
+    while (sel.length > 1) {
+        sel.removeChild(sel.lastChild);
+    }
+
+    //If we are only showing a single repo, we should have that option as selected
+    let selectedRepo: string
+    if (includedRepos.length == 1) {
+        selectedRepo = includedRepos[0].org.name + "/" + includedRepos[0].name
+    }
+
+    repos.forEach((repo) => {
+        const o = document.createElement("option");
+        o.text = repo;
+        o.value = "/configured-jobs/" + repo;
+        o.selected = repo === selectedRepo
+        sel.appendChild(o);
+    });
+
+    sel.addEventListener("change", (e) => {
+        window.location.href = sel.value;
+    })
+}
+
+/**
+ * Redraws the content of the provided repo.
+ */
+function redrawRepo(repo: Repo): void {
+    const container = document.querySelector("#job-container")!;
+    const repoContainer = container.querySelector(`#job-container-${repo.safeName}`)!;
+    while (repoContainer.childElementCount !== 0) {
+        repoContainer.removeChild(repoContainer.firstChild);
+    }
+
+    if (repo.jobs.length > 0) {
+        repo.jobs.forEach((job) => {
+            repoContainer.appendChild(createJobCard(job));
+        });
+    } else {
+        const message = document.createElement("h3");
+        message.innerHTML = "No Jobs found for " + repo.org.name + "/" + repo;
+        repoContainer.appendChild(message);
+    }
+}
+
+/**
+ * Creates and returns a card for the provided job
+ */
+function createJobCard(job: Job): HTMLElement {
+    const title = document.createElement("h3")
+    title.innerHTML = job.name;
+    title.classList.add("mdl-card__title-text");
+    const cardTitle = document.createElement("div");
+    cardTitle.classList.add("mdl-card__title");
+    cardTitle.appendChild(title);
+
+    const cardDesc = document.createElement("div");
+    cardDesc.innerHTML = job.type;
+    cardDesc.classList.add("mdl-card__supporting-text");
+
+    const cardAction = document.createElement("div");
+    const actionButton = document.createElement("a");
+    actionButton.innerHTML = "Details";
+    actionButton.classList.add(...["mdl-button", "mdl-button--colored", "mdl-js-button", "mdl-js-ripple-effect"]);
+    actionButton.addEventListener("click", () => {
+        const dialogElement = document.querySelector("dialog");
```

Comment:
I am not the master of ts, but should such statements be checked and error-ed if not found?

Reply:
I am also pretty green when it comes to ts, but this behavior was adapted from the following logic, and the "dialog" will never be not found unless someone introduces a bug with the HTML page. The bug would be painfully obvious, so I don't think we need additional error checking here.

https://github.com/kubernetes-sigs/prow/blob/739cbaba0899562e02585dc71587724e9b30c762/cmd/deck/static/plugin-help/plugin-help.ts#L30